scripts/pylib/twister/twisterlib/testinstance.py

```diff
@@ -204,8 +209,8 @@ def setup_handler(self, env):
     # Global testsuite parameters
     def check_runnable(self, enable_slow=False, filter='buildable', fixtures=[], hardware_map=None):
 
-        # running on simulators is currently not supported on Windows
-        if os.name == 'nt' and self.platform.simulation != 'na':
+        # running on simulators is currently supported only for QEMU on Windows
+        if os.name == 'nt' and self.platform.simulation not in ('na', 'qemu'):
```

Comment:
You described, that qemu has to be installed manually on windows. This means that by default it won't work. I don't see reflection of this in your implementation. E.g. for simulators that are not available on host by default it is first checked if the required components are available (in line 239). Can you add mechanism that will detect if qemu is available on Windows first, and only then don't skip such tests?
I am worried that without it it can brake existing setups and CI, where qemu tests will start failing instead of being skipped.

Reply:
no CI runs on windows, by now

Reply:
Not in the upstream, but other people might run such ones. E.g. we have one internally.

Reply:
@PerMac you are right, if QEMU is not present, test result is failed instead of skipped. A condition can be added to check if the QEMU executable is present: 
`        if os.name == 'nt' and not shutil.which(f"qemu-system-{self.platform.arch}"):
            return False`
but it is a bit tricky because qemu executables have a suffix given by architecture. For most platforms, creating the suffix from self.platform.arch works as shown in the code snippet above, but there are exceptions. For example, x86 translates to i386, and for some other platforms, the suffix depends on cmake a kconfig. The magic happens in qemu.cmake. The information is not available in self.platform. The simplest solution would be to check presence of one executable only, for example qemu-system-arm, and assume that all others are present as well. Would you agree with such solution, or do you have another idea?

Reply:
A reasonable solution can be to check presence of default QEMU installation path (C:\Program Files\qemu\) in PATH env variable and also check presence of QEMU_BIN_PATH env variable as it can point to another version of QEMU. The solution is not perfect as it does not check presence of a particular QEMU executable, but in my opinion, it serves the purpose well (purpose to skip tests on QEMU if it is not installed at all on a Windows host). Do you agree?